app/org/sagebionetworks/bridge/dao/HealthDataDao.java

```diff
@@ -29,6 +29,15 @@
     int deleteRecordsForHealthCode(@Nonnull String healthCode);
 
     /**
+     * DAO method used by worker apps to fetch a health data record by the record ID.
+     *
+     * @param id
+     *         record ID
+     * @return health data record
+     */
+    HealthDataRecord getRecordById(@Nonnull String id);
```

Comment:
Play doesn't route on query params, so I have both now routing to getRecord and I'm using a bunch of if statements to manually route on the query params.